builtin/mainmenu/tab_multiplayer.lua

```diff
@@ -199,16 +230,75 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		return true
 	end
 
+	if fields.btn_mp_search or fields.key_enter_field == "te_search" then
+		tabdata.fav_selected = 1
+		local input = fields.te_search:lower()
+		tabdata.search_for = fields.te_search
+
+		if #menudata.favorites < 2 then 
+			return true 
+		end
+
+		menudata.search_result = {}
+
+		-- setup the keyword list
+		local keywords = {}
```

Comment:
That would probably only make it the PR more complicated. I doubt there is a lot of code that can be used by both PR's.